### PR TITLE
feat: support auto create table config

### PR DESCRIPTION
**Pull request by MachaelLee**
## Which issue does this PR close?

Closes #

## Rationale for this change
 
Currently CeresDB create table automatically when writing data, if table does not exist, the time column is named as 'timestamp' which may not be not the same as user defined

## What changes are included in this PR?

Add config item of `auto_create_tables`

## Are there any user-facing changes?

User can configure `auto_create_tables' to allow the automatic creation of tables when writing data.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ target
 .DS_Store
 .idea/
 .vscode
+proto
```
Comment (chunshao90) [inline .gitignore:5]: Why ignore proto?

File: server/src/config.rs [hunk 2]
```diff
@@ -106,6 +106,9 @@ pub struct ServerConfig {
 
     /// Config for forwarding
     pub forward: forward::Config,
+
+    /// auto create tables, default true
+    pub auto_create_tables: bool,
```
Comment (chunshao90) [inline server/src/config.rs:111]: Rename to `auto_create_table`?

File: server/src/config.rs [hunk 3]
```diff
@@ -106,6 +106,9 @@ pub struct ServerConfig {
 
     /// Config for forwarding
     pub forward: forward::Config,
+
+    /// auto create tables, default true
```
Comment (jiacai2050) [inline server/src/config.rs:110]:
```suggestion
    /// Whether to create table when data is first written, only used in gRPC
```

File: server/src/grpc/storage_service/write.rs [hunk 4]
```diff
@@ -168,6 +169,7 @@ pub async fn execute_plan<Q: QueryExecutor + 'static>(
         })
 }
 
+#[allow(clippy::too_many_arguments)]
 pub async fn write_request_to_insert_plan<Q: QueryExecutor + 'static>(
```
Comment (jiacai2050) [inline server/src/grpc/storage_service/write.rs:173]: I think we can add a `WriteContext` to wrap args here:
```rs
pub struct WriteContext {
    pub request_id: RequestId,
    pub deadline: Option<Instant>,
    pub catalog: String,
    pub schema: String,
    pub auto_create_tables: bool
}

```

File: .gitignore (hunk 1 above)
Comment (MachaelLee) [reply .gitignore:5]: I will remove it.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**